src/bin/pg_autoctl/keeper.c

```diff
@@ -1402,21 +1423,22 @@ keeper_check_monitor_extension_version(Keeper *keeper)
 	}
 
 	/* from a member of the cluster, we don't try to upgrade the extension */
-	if (strcmp(version.installedVersion, PG_AUTOCTL_EXTENSION_VERSION) != 0)
+	if (strcmp(version->installedVersion, PG_AUTOCTL_EXTENSION_VERSION) != 0)
 	{
 		log_fatal("The monitor at \"%s\" has extension \"%s\" version \"%s\", "
 				  "this pg_autoctl version requires version \"%s\".",
 				  keeper->config.monitor_pguri,
 				  PG_AUTOCTL_MONITOR_EXTENSION_NAME,
 				  PG_AUTOCTL_EXTENSION_VERSION,
-				  version.installedVersion);
+				  version->installedVersion);
 		log_info("Please connect to the monitor node and restart pg_autoctl.");
```

Comment:
I think the message is meant for the situation where the client has been upgraded (say, to 1.6) already when the monitor is still running 1.5 ; and so it would mean that you need to restart the monitor. That said, I agree that it's confusing and not helpful, so I have removed it.